src/GoTrueClient.ts

```diff
@@ -588,73 +636,70 @@ export default class GoTrueClient {
     }
   }
 
-  private _doesSessionExist(maybeSession: unknown): maybeSession is Session {
-    const isValidSession =
-      typeof maybeSession === 'object' &&
-      maybeSession !== null &&
-      'access_token' in maybeSession &&
-      'refresh_token' in maybeSession &&
-      'expires_at' in maybeSession
-
-    return isValidSession
-  }
-
-  private async _handleEmailSignIn(
+  /**
+   * Sends a reset request to an email address.
+   * @param email The email address of the user.
+   * @param options.redirectTo A URL to send the user to after they are confirmed.
+   * @param options.captchaToken Verification token received when the user completes the captcha on the site.
+   */
+  async resetPasswordForEmail(
     email: string,
-    password: string,
     options: {
+      redirectTo?: string
       captchaToken?: string
     } = {}
-  ): Promise<AuthResponse> {
+  ): Promise<
+    | {
+        data: {}
+        error: null
+      }
+    | { data: null; error: AuthError }
+  > {
     try {
-      const { data, error } = await this.api.signInWithEmail(email, password, {
-        captchaToken: options.captchaToken,
+      return await _request(this.fetch, 'POST', `${this.url}/recover`, {
+        body: { email, gotrue_meta_security: { captcha_token: options.captchaToken } },
+        headers: this.headers,
+        redirectTo: options.redirectTo,
       })
-      if (error || !data) return { data: { user: null, session: null }, error }
-
-      if (data?.user?.confirmed_at || data?.user?.email_confirmed_at) {
-        this._saveSession(data)
-        this._notifyAllSubscribers('SIGNED_IN', data)
-      }
-
-      return { data: { user: data.user, session: data }, error: null }
     } catch (error) {
       if (isAuthError(error)) {
-        return { data: { user: null, session: null }, error }
+        return { data: null, error }
       }
 
       throw error
     }
   }
 
-  private async _handlePhoneSignIn(
-    phone: string,
-    password: string,
-    options: {
-      captchaToken?: string
-    } = {}
-  ) {
+  /**
+   * Generates a new JWT.
+   * @param refreshToken A valid refresh token that was returned on login.
+   */
+  async refreshAccessToken(refreshToken: string): Promise<AuthResponse> {
```

Comment:
@alaister should this be made private too? seems like it can be handled by `setSession` which also saves the session

Reply:
This one I think we should leave public as users may want to manually refresh a session if they've updated the user info server-side for example

Reply:
Can you remember why we had to pass a `refreshToken` to it, though, instead of using the one in the session?

Reply:
decided to make it private since we have `setSession` does basically the same thing but also updates the session stored in memory / in the browser